src/backend/aspen/workflows/nextclade/save.py

```diff
@@ -13,117 +14,235 @@
     SqlAlchemyInterface,
 )
 from aspen.database.models import (
-    Group,
     LineageType,
+    MutationsCaller,
+    QCMetricCaller,
     Sample,
     SampleLineage,
     SampleMutation,
     SampleQCMetric,
 )
 
+# TODO, create an enum table for below and standard nextclade QC overallStatus
+INVALID_RESULT_STATUS = "invalid"
+
 
 @click.command("save")
 @click.option("nextclade_fh", "--nextclade-csv", type=click.File("r"), required=True)
+@click.option(
+    "nextclade_tag_fh", "--nextclade-dataset-tag", type=click.File("r"), required=True
+)
 @click.option("nextclade_version", "--nextclade-version", type=str, required=True)
-@click.option("group_name", "--group-name", type=str, required=True)
 @click.option("pathogen_slug", "--pathogen-slug", type=str, required=True)
 def cli(
     nextclade_fh: io.TextIOBase,
+    nextclade_tag_fh: IO[str],
     nextclade_version: str,
-    group_name: str,
     pathogen_slug: str,
 ):
-    interface: SqlAlchemyInterface = init_db(get_db_uri(Config()))
+    """Go through results from nextclade run, save to DB for each sample."""
+    # Track info about the dataset that was used to produce results being saved
+    dataset_info = extract_dataset_info(nextclade_tag_fh)
 
+    interface: SqlAlchemyInterface = init_db(get_db_uri(Config()))
     with session_scope(interface) as session:
         nextclade_csv: csv.DictReader = csv.DictReader(nextclade_fh, delimiter=";")
         for row in nextclade_csv:
-            sample_q = (
-                sa.select(Sample)
-                .join(Sample.submitting_group)
-                .filter(
-                    and_(
-                        Sample.public_identifier == row["seqName"],
-                        Group.name == group_name,
-                    )
-                )
-            )
+            # For entire workflow, we use sample id primary keys for names.
+            sample_id = int(row["seqName"])
+            sample_q = sa.select(Sample).where(Sample.id == sample_id)
             sample = session.execute(sample_q).scalars().one()
+
+            is_result_valid = is_nextclade_result_valid(row)
+            # We always record QC info for any sample run, even if invalid.
+            qc_score = row["qc.overallScore"]
+            qc_status = row["qc.overallStatus"]
+            if not is_result_valid:
+                # HACK Setting `qc_score` to empty string is a bit of an abuse.
+                # DB has that field as a non-nullable string, but Nextclade's
+                # qc.overallScore is actually a number. We store CSV text of
+                # that number, so this is a compromise to indicate no result.
+                qc_score = ""
+                qc_status = INVALID_RESULT_STATUS
```

Comment:
we could just change the qc_score to be nullable?

Reply:
Yeah, if you think that's the way to go, I'm happy to go make it nullable. I think you and Jess were both out when I wrote that, and I wasn't sure if there was a strong need for `qc_score` to not be nullable, so I just did that in the meantime and forgot to ask when you were back.